src/plugins/image_embeddable/public/image_editor/image_editor_flyout.tsx

```diff
@@ -0,0 +1,461 @@
+import { validateImageConfig, DraftImageConfig } from '../utils/validate_image_config';
+
+/**
+ * Shared sizing css for image, upload placeholder, empty and not found state
+ * Makes sure the container has not too large height to preserve vertical space for the image configuration in the flyout
+ */
+const CONTAINER_SIZING_CSS = css({
+  aspectRatio: `21 / 9`,
+  width: `100%`,
+  height: `auto`,
+  maxHeight: `max(20vh, 180px)`,
+});
+
+export interface ImageEditorFlyoutProps {
+  onCancel: () => void;
+  onSave: (imageConfig: ImageConfig) => void;
+  initialImageConfig?: ImageConfig;
+  validateUrl: ValidateUrlFn;
+}
+
+export function ImageEditorFlyout(props: ImageEditorFlyoutProps) {
+  const { euiTheme } = useEuiTheme();
+  const [fileId, setFileId] = useState<undefined | string>(() =>
+    props.initialImageConfig?.src?.type === 'file' ? props.initialImageConfig.src.fileId : undefined
+  );
+  const [fileImageMeta, setFileImageMeta] = useState<undefined | FileImageMetadata>(() =>
+    props.initialImageConfig?.src?.type === 'file'
+      ? props.initialImageConfig.src.fileImageMeta
+      : undefined
+  );
+  const [srcType, setSrcType] = useState<ImageConfig['src']['type']>(
+    () => props.initialImageConfig?.src?.type ?? 'file'
+  );
+  const [srcUrl, setSrcUrl] = useState<string>(() =>
+    props.initialImageConfig?.src?.type === 'url' ? props.initialImageConfig.src.url : ''
+  );
+  const [srcUrlError, setSrcUrlError] = useState<string | null>(() => {
+    if (srcUrl) return props.validateUrl(srcUrl)?.error ?? null;
+    return null;
+  });
+  const [isFilePickerOpen, setIsFilePickerOpen] = useState<boolean>(false);
+  const [sizingObjectFit, setSizingObjectFit] = useState<ImageConfig['sizing']['objectFit']>(
+    () => props.initialImageConfig?.sizing?.objectFit ?? 'contain'
+  );
+  const [altText, setAltText] = useState<string>(() => props.initialImageConfig?.altText ?? '');
+  const [color, setColor, colorErrors] = useColorPickerState(
+    props?.initialImageConfig?.backgroundColor
+  );
+  const isColorInvalid = !!color && !!colorErrors;
+
+  const draftImageConfig: DraftImageConfig = {
+    ...props.initialImageConfig,
+    src:
+      srcType === 'url'
+        ? {
+            type: 'url',
+            url: srcUrl,
+          }
+        : { type: 'file', fileId, fileImageMeta },
+    altText,
+    backgroundColor: colorErrors ? undefined : color,
+    sizing: {
+      objectFit: sizingObjectFit,
+    },
+  };
+
+  const isDraftImageConfigValid = validateImageConfig(draftImageConfig, {
+    validateUrl: props.validateUrl,
+  });
+
+  const onSave = () => {
+    if (!isDraftImageConfigValid) return;
+    props.onSave(draftImageConfig);
+  };
+
+  return (
+    <>
+      <EuiFlyoutHeader hasBorder={true}>
+        <EuiTitle size="m">
+          <h2>
+            <FormattedMessage
+              id="imageEmbeddable.imageEditor.title"
+              defaultMessage="Configure Image"
```

Comment:
> Suggest using "Add image" for the title

👍 (also "Edit image" when editing) 

> for the tooltip on the image icon.

Unfortunately this one is tricky because there are a lot of indirection in code. leaving untouched